src/solvers/petsc_nonlinear_solver.C

```diff
@@ -715,6 +715,11 @@ PetscNonlinearSolver<T>::solve (SparseMatrix<T> &  pre_in,  // System Preconditi
       LIBMESH_CHKERR(ierr);
       ierr = MatMFFDSetFunction(J, __libmesh_petsc_snes_mffd_interface, this);
       LIBMESH_CHKERR(ierr);
+#if !PETSC_VERSION_LESS_THAN(3, 8, 3)
```

Comment:
If this was a recent change, it won't be in 3.8.3. 

We'll probably have to test `!PETSC_VERSION_LESS_THAN(3, 8, 4)` if you are sure it will be in the next patch level release...

Reply:
Let us hold this PR until PETSc-side is done.